src/transformers/models/maskformer/modeling_maskformer.py

```diff
@@ -0,0 +1,1638 @@
+            window_size=config.swin_window_size,
+        )
+        self.pixel_decoder = MaskFormerPixelDecoder(
+            in_features=self.backbone.get_outputs_shape()[-1],
+            feature_size=config.fpn_feature_size,
+            mask_feature_size=config.mask_feature_size,
+            lateral_widths=self.backbone.get_outputs_shape()[:-1],
+        )
+
+    def forward(self, x: Tensor) -> Tuple[Tensor, Tensor]:
+        features: List[Tensor] = self.backbone(x)
+        # the last feature is actually the output from the last layer
+        image_features: Tensor = features[-1]
+        pixel_embeddings: Tensor = self.pixel_decoder(features)
+        return image_features, pixel_embeddings
+
+
+class MaskFormerTransformerModule(nn.Module):
+    def __init__(self, in_features: int, config: MaskFormerConfig):
+        super().__init__()
+        self.position_embedder = PositionEmbeddingSine(num_pos_feats=config.hidden_size // 2, normalize=True)
+        self.queries_embedder = nn.Embedding(config.num_queries, config.hidden_size)
+        should_project = in_features != config.hidden_size
+        self.input_proj = (
+            nn.Conv2d(in_features, config.hidden_size, kernel_size=1) if should_project else nn.Identity()
+        )
+        # TODO hugly, ask!
+        self.detr_decoder = DetrDecoder(
+            DetrConfig(**{k: v for k, v in config.__dict__.items() if k.startswith("detr_")})
+        )
+
+    def forward(self, image_features: Tensor) -> Tuple[Tensor]:
+        image_features = self.input_proj(image_features)
+        position_embeddings: Tensor = self.position_embedder(image_features)
+        queries_embeddings: Tensor = repeat(self.queries_embedder.weight, "q c -> b q c", b=image_features.shape[0])
+        inputs_embeds = torch.zeros_like(queries_embeddings)
+        image_features = rearrange(image_features, "b c h w -> (h w) b c")
+        position_embeddings = rearrange(position_embeddings, "b c h w -> (h w) b c")
+
+        detr_output: DetrDecoderOutput = self.detr_decoder(
+            inputs_embeds=inputs_embeds,
+            attention_mask=None,
+            encoder_hidden_states=image_features,
+            encoder_attention_mask=None,
+            position_embeddings=position_embeddings,
+            query_position_embeddings=queries_embeddings,
+            output_attentions=None,
+            output_hidden_states=True,
+            return_dict=None,
+        )
+        # TODO return a tuple is not a good idea, torch.summary will fail
+        return detr_output.hidden_states
+
+
+class MaskFormerSegmentationModule(nn.Module):
+    def __init__(self, config: MaskFormerConfig):
+        super().__init__()
+        # + 1 because we add the "null" class
+        self.mask_classification = config.mask_classification
+        self.class_predictor = nn.Linear(config.hidden_size, config.num_classes + 1)
+        self.mask_embedder = MLP(config.hidden_size, config.hidden_size, config.mask_feature_size)
```

Comment:
```suggestion
        self.mask_embedder = MaskformerMLPPredictionHead(config.hidden_size, config.hidden_size, config.mask_feature_size)
```
If we are going with the name `MaskformerMLPPredictionHead`, we should also use it here.